File: Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace hotel_management.Models
{
    public class Booking
    {
        public int Id { get; set; }

        public int RoomId { get; set; }
        public Room Room { get; set; }

        [Required]
        public string GuestName { get; set; }

        [Required]
        public DateTime CheckIn { get; set; }

        [Required]
        public DateTime CheckOut { get; set; }

        public decimal TotalPrice { get; set; }

        public string Status { get; set; }
    }
}

File: lib/models/booking.dart

class Booking {
  final String id;
  final String roomId;
  final String customerId;
  final DateTime checkInDate;
  final DateTime checkOutDate;
  final double totalPrice;
  final String status; // 'Active', 'Completed', 'Cancelled'

  Booking({
    required this.id,
    required this.roomId,
    required this.customerId,
    required this.checkInDate,
    required this.checkOutDate,
    required this.totalPrice,
    this.status = 'Active',
  });
} 